src/components/dialog/dialog.spec.tsx

```diff
@@ -776,3 +817,26 @@ describe("Dialog", () => {
     );
   });
 });
+describe("when imperative handle is forwarded to Dialog", () => {
+  it("calling handle's focus method refocuses on Dialog's root container", async () => {
+    const MockComponent = () => {
+      const dialogHandle = React.useRef<DialogHandle>(null);
+
+      return (
+        <Dialog open title="My dialog" ref={dialogHandle}>
+          <Button onClick={() => dialogHandle.current?.focus()}>
+            Press me to refocus on Dialog
+          </Button>
+        </Dialog>
+      );
+    };
+    const user = userEvent.setup();
+    render(<MockComponent />);
+    const button = screen.getByRole("button");
+    button.focus();
+
+    await user.keyboard("[Enter]");
```

Comment:
**comment** (non-blocking): it doesn't really matter, but I'm curious that you've chosen to trigger an Enter press rather a Click on the button in order to trigger the focus. It works fine because of course an enter press on an HTML button triggers the click event, but that isn't what we're actually testing here so I'd prefer to keep it simple and just trigger the click event directly. (And there's no need to test keyboard behaviour separately, as that's the responsibility of Button rather than anything to do with Dialog.)

Reply:
Good point 👍🏼 Previously instead of manually focusing on the button on line 836, I tried to instead simulate pressing tab to focus on the button as its more inline with what the user would do. However, I ran into some issues with the focus trap which made the test flakey, so opted to manually set focus instead.

Given I've done this, I'm happy to change the action to a click 👍🏼